src/nodes/chunk_append/chunk_append.c

```diff
@@ -319,21 +319,31 @@ ts_chunk_append_path_create(PlannerInfo *root, RelOptInfo *rel, Hypertable *ht,
 			if (flat == NULL)
 				break;
 
+			/*
+			 * For each lc_oid, there will be 0, 1, or 2 matches in flat_list: 0 matches
+			 * if child was pruned, 1 match if the chunk is uncompressed or fully compressed,
+			 * 2 matches if the chunk is partially compressed.
+			 * If there are 2 matches they will also be consecutive (see assumption above)
+			 */
 			foreach (lc_oid, current_oids)
 			{
-				/* postgres may have pruned away some children already */
-				Path *child = (Path *) lfirst(flat);
-				Oid parent_relid = child->parent->relid;
-				bool is_not_pruned =
-					lfirst_oid(lc_oid) == root->simple_rte_array[parent_relid]->relid;
-
-				if (is_not_pruned)
+				bool is_not_pruned = true;
+				do
 				{
-					merge_childs = lappend(merge_childs, child);
-					flat = lnext_compat(children, flat);
-					if (flat == NULL)
-						break;
-				}
+					Path *child = (Path *) lfirst(flat);
+					Oid parent_relid = child->parent->relid;
+					is_not_pruned =
+						lfirst_oid(lc_oid) == root->simple_rte_array[parent_relid]->relid;
+					/* postgres may have pruned away some children already */
+					if (is_not_pruned)
+					{
+						merge_childs = lappend(merge_childs, child);
+						flat = lnext_compat(children, flat);
+						if (flat == NULL)
+							break;
```

Comment:
What would that mean, nested_oids and path children not in sync?

Reply:
I think it means that some of the children we identified in `nested_oids` as part of a chunk to be included have actually been “pruned” by postgres and thus not included in the flat list, which contains the mergeAppend children. But it seems we expect this possibility (for these lists to differ) so I don’t know if that would make them out of sync. However, I don’t know yet in which scenario postgres would have pruned some of the children, so that’s something for me to figure out yet.